### PR TITLE
8260245: Update ICU4C to version 68.2

### PR DESCRIPTION
**Pull request by arun-Joseph**
We currently use ICU4C version 64.2. We should update to the latest stable version 68.2.
http://site.icu-project.org/home

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issues
 * [JDK-8260245](https://bugs.openjdk.java.net/browse/JDK-8260245): Update ICU4C to version 68.2
 * [JDK-8232379](https://bugs.openjdk.java.net/browse/JDK-8232379): Need to remove large icudt64l.zip binary file from source repository

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Johan Vos](https://openjdk.java.net/census#jvos) (@johanvos - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/456/head:pull/456` \
`$ git checkout pull/456`

Update a local copy of the PR: \
`$ git checkout pull/456` \
`$ git pull https://git.openjdk.java.net/jfx pull/456/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 456`

View PR using the GUI difftool: \
`$ git pr show -t 456`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/456.diff">https://git.openjdk.java.net/jfx/pull/456.diff</a>

</details>

### CODE REVIEW

**Comment by kevinrushforth** [comment]
Overall this looks good. I'll review / test it in detail later.

Quick comments on the PR:

1. You need to update the version number in `.../legal/icu_web.md`
2. Related to the above, can you create a new `UPDATING.txt` file in `.../native/Source/ThirdParty/icu` with instructions as to what needs to be updated when updating new ICU files, specifically including the changes to `build.gradle`, `.../legal/icu_web.md`, and `/native/Source/ThirdParty/icu/CMakeLists.txt` regarding updating the version number, along with anything else that isn't obvious (although I think the rest is straight-forward)?
3. Will the changes in core WebKit code (e.g., `TRUE` --> `true`) cause merge conflicts in the future?

Also, since this also addresses [JDK-8232379](https://bugs.openjdk.java.net/browse/JDK-8232379), can you add it to this PR with `/issue add 8232379`?

/reviewers 2

**Comment by arun-Joseph** [comment]
The other WebKit changes are present in the upstream repo as well (https://bugs.webkit.org/show_bug.cgi?id=218522).

File: build.gradle [hunk 1]
```diff
@@ -1902,6 +1903,20 @@ allprojects {
...
 
+    if (JFX_DEPS_URL == "" && IS_COMPILE_WEBKIT) {
+        repositories {
+            ivy {
+                url icuRepositoryURL
+                metadataSources {
+                    artifact()
+                }
+                patternLayout {
+                    artifact "download/release-68-2/[artifact].[ext]"
```
Comment (johanvos) [inline build.gradle:1914]: shouldn't 68-2 be a property?

File: build.gradle [hunk 2]
```diff
@@ -3362,6 +3379,9 @@ project(":web") {
     commonModuleSetup(project, [ 'base', 'graphics', 'controls', 'media', 'web' ])
 
     dependencies {
+        if (IS_COMPILE_WEBKIT) {
+            icu name: "icu4c-68.2-data-bin-l", ext: "zip"
```
Comment (kevinrushforth) [inline build.gradle:3383]: I think the `68.2` could also be a property.

**Review by kevinrushforth — COMMENTED** [review]
I did a review of all change not in the ThirdParty/icu. All looks good to me. There is one more instance of the hard-coded `68.2` that could be defined as a property.

I've done sanity testing on Windows and and Mac. No problems that I can see. I'll do a bit more testing.

**Review by johanvos — APPROVED** [review]

**Review by kevinrushforth — APPROVED** [review]
Testing looks good (all platforms). So do the changes.

**Pull request closed without merging**